src/submission-handlers/obsHandler.ts

```diff
@@ -0,0 +1,231 @@
+      );
+    }
+    if (rendering === 'toggle') {
+      return value ? field.questionOptions.toggleOptions.labelTrue : field.questionOptions.toggleOptions.labelFalse;
+    }
+    if (codedTypes.includes(rendering)) {
+      return field.questionOptions.answers?.find((option) => option.concept == value)?.label;
+    }
+    return value;
+  },
+  getPreviousValue: (field: FormField, encounter: OpenmrsEncounter, allFormFields: Array<FormField>) => {
+    return extractFieldValue(field, findObsByFormField(flattenObsList(encounter.obs), assignedObsIds, field), false);
+  },
+};
+
+// Helpers
+
+/**
+ * Extracts field's primitive value from obs
+ */
+function extractFieldValue(field: FormField, obsList: OpenmrsObs[] = [], makeFieldDirty = false) {
+  const rendering = field.questionOptions.rendering;
+  if (!field.meta) {
+    field.meta = {
+      previousValue: null,
+    };
+  }
+  if (obsList.length) {
+    if (rendering == 'checkbox') {
+      assignedObsIds.push(...obsList.map((obs) => obs.uuid));
+      field.meta.previousValue = makeFieldDirty ? obsList : null;
+      return obsList.map((o) => o.value.uuid);
+    }
+    const obs = obsList[0];
+    if (makeFieldDirty) {
+      field.meta.previousValue = { ...obs };
+    }
+    assignedObsIds.push(obs.uuid);
+    if (typeof obs.value === 'string' || typeof obs.value === 'number') {
+      if (rendering.startsWith('date')) {
+        const dateObject = parseToLocalDateTime(obs.value as string);
+        if (makeFieldDirty) {
+          field.meta.previousValue.value = dayjs(dateObject).format('YYYY-MM-DD HH:mm');
+        }
+        return dateObject;
+      }
+      return obs.value;
+    }
+    if (rendering == 'toggle') {
+      return obs.value.uuid === ConceptTrue;
+    }
+    if (rendering == 'fixed-value') {
+      return field['fixedValue'];
+    }
+    return obs.value?.uuid;
+  }
+  return '';
+}
+
+export function constructObs(field: FormField, value: any) {
+  if (isEmpty(value) && field.type !== 'obsGroup') {
+    return null;
+  }
+  const draftObs =
+    field.type === 'obsGroup'
+      ? { groupMembers: [] }
+      : {
+          value: field.questionOptions.rendering.startsWith('date') ? formatDate(field, value) : value,
+        };
+  return {
+    ...draftObs,
+    concept: field.questionOptions.concept,
+    formFieldNamespace: 'rfe-forms',
+    formFieldPath: `rfe-forms-${field.id}`,
+  };
+}
+
+export function voidObs(obs: OpenmrsObs) {
+  return { uuid: obs.uuid, voided: true };
+}
+
+function editObs(field: FormField, newValue: any) {
+  const oldObs = field.meta.previousValue;
+  const formatedValue = field.questionOptions.rendering.startsWith('date') ? formatDate(field, newValue) : newValue;
+  return {
+    uuid: oldObs.uuid,
+    value: formatedValue,
+    formFieldNamespace: 'rfe-forms',
+    formFieldPath: `rfe-forms-${field.id}`,
+  };
+}
+
+function formatDate(field: FormField, value: any) {
```

Comment:
Is value a string here? If not, it doesn't look like we're doing anything special to make it a string. Probably safe to assume it is a string then and annotate it appropriately.